modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java

```diff
@@ -4,314 +4,120 @@
-                if (filteredTags.contains(it.next().getName())) {
-                    it.remove();
-                }
-            }
-            if (clone.getTags().isEmpty()) {
-                clone.setTags(null);
-            }
-        }
-
-        Map<String, Schema> definitions = filterDefinitions(filter, swagger.getComponents().getSchemas(), params, cookies, headers);
-        // TODO
-//        clone.setSecurityDefinitions(swagger.getSecurityDefinitions());
-        clone.setSecurity(swagger.getSecurity());
-        // TODO
-//        clone.setDefinitions(definitions);
-
-        // isRemovingUnreferencedDefinitions is not defined in SwaggerSpecFilter to avoid breaking compatibility with
-        // existing client filters directly implementing SwaggerSpecFilter.
-        if (filter instanceof AbstractSpecFilter) {
-            if (((AbstractSpecFilter)filter).isRemovingUnreferencedDefinitions()) {
-                clone = removeBrokenReferenceDefinitions (clone);
+        OpenAPI clone = new OpenAPI();
+        clone.info(openAPI.getInfo());
+        clone.openapi(openAPI.getOpenapi());
+        clone.tags(openAPI.getTags() == null ? null : new ArrayList<>(openAPI.getTags()));
+
+        Paths clonedPaths = new Paths();
+        for (String resourcePath : openAPI.getPaths().keySet()) {
+            PathItem pathItem = openAPI.getPaths().get(resourcePath);
+
+            PathItem filteredPathItem = filterPathItem(filter, pathItem, resourcePath, null, params, cookies, headers);
+            if (filteredPathItem != null) {
+                filteredPathItem.setGet(filterOperation(filter, pathItem.getGet(), resourcePath, GET, params, cookies, headers));
+                filteredPathItem.setPost(filterOperation(filter, pathItem.getPost(), resourcePath, POST, params, cookies, headers));
+                filteredPathItem.setPut(filterOperation(filter, pathItem.getPut(), resourcePath, PUT, params, cookies, headers));
+                filteredPathItem.setDelete(filterOperation(filter, pathItem.getDelete(), resourcePath, DELETE, params, cookies, headers));
+                filteredPathItem.setPatch(filterOperation(filter, pathItem.getPatch(), resourcePath, PATCH, params, cookies, headers));
+                filteredPathItem.setHead(filterOperation(filter, pathItem.getHead(), resourcePath, HEAD, params, cookies, headers));
+                filteredPathItem.setOptions(filterOperation(filter, pathItem.getOptions(), resourcePath, OPTIONS, params, cookies, headers));
+                filteredPathItem.setDescription(pathItem.getDescription());
+                filteredPathItem.set$ref(pathItem.get$ref());
+                filteredPathItem.setSummary(pathItem.getSummary());
+                clonedPaths.addPathItem(resourcePath, filteredPathItem);
             }
+            clone.paths(clonedPaths);
+            clone.setComponents(openAPI.getComponents());
+            clone.setSecurity(openAPI.getSecurity());
```

Comment:
these should be called outside the loop

Reply:
also other elements are not cloned? e.g. extensions, externalDocs, servers

Reply:
Done!